specs/commandline/key.md

```diff
@@ -0,0 +1,131 @@
+# notation key
+
+## Description
+
+Use ```notation key``` command to manage keys used for signing. User can add/update/list/remove key to/from signing key list. Please be noted this command doesn't manage the lifecycle of signing key itself, it manages the signing key list only.
+
+## Outline
+
+### notation key command
+
+```text
+Manage keys used for signing
+
+Usage:
+  notation key [command]
+
+Available Commands:
+  add         Add key to signing key list
+  list        List keys used for signing
+  remove      Remove key from signing key list
+  update      Update key in signing key list
+
+Flags:
+  -h, --help   help for key
+```
+
+### notation key add
+
+```text
+Add key to signing key list
+
+Usage:
+  notation key add <key_name> [key_path cert_path] [flags]
+
+Flags:
+  -d, --default               mark as default
+  -h, --help                  help for add
+      --id string             key id (required if --plugin is set)
+  -p, --plugin string         signing plugin name
+      --plugin-config string  list of comma-separated {key}={value} pairs that are passed as is to the plugin, refer plugin documentation to set appropriate values
```

Comment:
It is better to not have `comma-separated {key}={value} pairs` since `value` itself MAY have `,`.
```suggestion
      --plugin-config strings {key}={value} pairs that are passed as is to the plugin, refer plugin documentation to set appropriate values
```

For example,
```shell
notation key add -p test --plugin-config foo=fun --plugin-config bar=beer test_key
```